### DISCUSSION TITLE
Issue connecting Project CalDav with Hetzner Webmail

### DISCUSSION THREAD

**Discussion by Noctran** [super-productivity/super-productivity#3056] [Q&A] [answered]
I'm trying to have one Project sync with the webmail Account I have with [hetzner](https://www.hetzner.com/webhosting/) via CalDav. This already works with Thunderbird for example but there trough login with the associated email.

[Hetzners documentation](https://docs.hetzner.com/konsoleh/account-management/email/webmail-faq) and in App Ressources show the connection string to be: `https://webmail.your-server.de/rpc/calendars/(email)/tasks~(random character string)/`.
I have tried every combination of base URL and calender that would have made sense to me but I can't get SP to connect to the Remote Source.

**Comment by hdokes** [comment]
Hello Noctran,

Your inquiry regarding a CalDav connection to Hetzner Webmail is not unlike our recent inquiry for issues we are experiencing creating a CalDav connection to MailCow.... thread #3064

Open a Developer Tools from within SP (super-productivity) and while in the setup screen for CalDav make your adjustments there to the BaseURL and the recource then look at the network tab of the Developer Tools and see what return code you are getting as you make an adjustment to the path.  If it is in the 200's than you are successfully connecting and the issue is beyond that connection.  It may provide you a different code in the 400's for example that will reveal it actually is a connection issue and narrow you down to why.

We are actually making connection to CalDav in MailCow but the information being returned is not parsing properly leading to a code error trying to process a null return.

I invite you to look for those return codes and monitor our thread as well and perhaps we will both find our solution.

**Comment by Noctran** [comment, marked as answer]
For anybody else using Hetzner encountering this Issue:
Tasks -> Edit Tasklist (Pen symbol in the "My Task Lists" Listing) -> CalDAV Subscription URL without the "tasks~{ID}" Part as URL & and the Content of the top most Field as the Name

---

I have, while writing this comment because I hadn't found a solution, found a Solution xD

The Hetzner Webmail Interface offers multiple subscription String as well as separate CalDav Connection data for Calendar and Tasks.
![image](https://github.com/user-attachments/assets/7d575343-a3d6-435b-adf5-37709c2c364e)

I tried every conceivable Combination of:

CalDAV Subscription URL Main: "https://webmail.your-server.de/rpc/calendars/{account mail}/"
CalDAV Calendar ID: "calendar~{ID String}"
CalDAV Account URL: "https://webmail.your-server.de/rpc/principals/{account mail}"

As well as with the separate Tasks Connection strings.

Only for this to be the Solution:
![image](https://github.com/user-attachments/assets/0ffff486-c397-4b6d-a927-dd36616164fb)

*This successfully connects, but at least in the Case of Hetzner, won't sync your Tasks, you have to use the equivalent Tasks Information for that

So technically, it's not really wrong, I just didn't consider that top field for Calendar Title technical information. More like something, an Alias to make the differentiation between the actual IDs

@janLo sorry if pinging you is inappropriate, but I'm unsure if this is the intended behavior, so wanted to make sure you know about it in case it's not. 
(And if it is, maybe this could be added to the help text or something, at least if Hetzner isn't the only provider like this) 
(if you want / would benefit from an Account with Hetzner for testing or something, I can create a separate Account with one of my domains for that use for you, without any extra cost:))

@hdokes just in case this can be useful for your Issue:)

**Reply by Asqiir** [reply]
with this I can create tasks, yay!

But I can't create new lists, is this normal or an issue on my side somehow?
